### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by willtebbutt**
**Summary**

The priorities seemed a bit out of date and, given that we don't really have a coherant list of priorities for this package at the minute, I figured it best to remove them for now.

In their place I propose a related work section.

**Proposed changes**

- Remove priority list in README
- Add related work section to README

**What alternatives have you considered?**

Trying to update the priorities to reflect where we are today. This turned out not to be possible, owing to a lack of a clear set of priorities.

**Breaking changes**

None

### CODE REVIEW

**Review by st-- — APPROVED** [review]
Seems sensible!

Would you want to keep a note that KernelFunctions is and will be AD-compatible though?

**Comment by willtebbutt** [comment]
> Would you want to keep a note that KernelFunctions is and will be AD-compatible though?

Good point. Yes, I want that.

File: README.md [hunk 1]
```diff
@@ -12,10 +12,12 @@
 
 ## Kernel functions for machine learning
 
-KernelFunctions.jl provide a flexible and complete framework for kernel functions, pretransforming the input data.
+KernelFunctions.jl provide a flexible and complete framework for kernel functions.
```
Comment (st--) [inline README.md:15]:
```suggestion
KernelFunctions.jl provides a flexible and complete framework for kernel functions.
```
nb- "complete" in what sense ?

**Comment by willtebbutt** [comment]
@theogf @st-- added AD reference and removed reference to input transformations, because I'm not sure that input transformations will mean much to users first encountering the package. Happy to revert if the latter change is at all contentious.

File: README.md (hunk 1 above)
Comment (willtebbutt) [reply README.md:15]: A good question. I suspect @theogf wrote this originally, so probably the person to ask.

File: README.md [hunk 2]
```diff
@@ -12,10 +12,12 @@
...
 
 The aim is to make the API as model-agnostic as possible while still being user-friendly.
```
Comment (st--) [inline README.md:17]: While we're looking at it: should we mention here that KernelFunctions.jl does not handle parameters by itself (ensuring constraints such as positivity, or getting its gradients) and point to other packages such as ParameterHandling.jl and Flux's parameters for how to do that?

File: README.md (hunk 1 above)
Comment (theogf) [reply README.md:15]: 🤷

File: README.md (hunk 2 above)
Comment (Crown421) [reply README.md:17]: I was planning to add an example on parameter handling to the docs over the next few days. I have been experimenting, and should be easy to clean it up and make a PR.

File: README.md (hunk 1 above)
Comment (theogf) [reply README.md:15]: I think I mean `extensive` or some word meaning that we have a lot of features!

Comment (willtebbutt) [reply README.md:15]: How about something like
```suggestion
KernelFunctions.jl provides a flexible framework for defining kernel functions, and an extensive collection of implementations.
```

File: README.md (hunk 2 above)
Comment (willtebbutt) [reply README.md:17]: > I was planning to add an example on parameter handling to the docs over the next few days. I have been experimenting, and should be easy to clean it up and make a PR.

Nice. I look forward to it

> While we're looking at it: should we mention here that KernelFunctions.jl does not handle parameters by itself (ensuring constraints such as positivity, or getting its gradients) and point to other packages such as ParameterHandling.jl and Flux's parameters for how to do that?

Good point. Would something like
```suggestion
The aim is to make the API as model-agnostic as possible while still being user-friendly, and to interoperate well with generic packages for handling parameters like ParameterHandling.jl and Functors.jl.
```
be what you're after @st-- ?

Comment (theogf) [reply README.md:17]: Looks great! Maybe you could also include the links to the packages

Comment (willtebbutt) [reply README.md:17]:
```suggestion
The aim is to make the API as model-agnostic as possible while still being user-friendly, and to interoperate well with generic packages for handling parameters like [ParameterHandling.jl](https://github.com/invenia/ParameterHandling.jl/) and FluxML's [Functors.jl](https://github.com/FluxML/Functors.jl/).
```

**Comment by willtebbutt** [comment]
@theogf @st-- happy with the state of this now?

File: README.md [hunk 3]
```diff
@@ -46,12 +48,12 @@ plot(
...
 </p>
 
-## Packages goals (by priority)
-- Ensure AD Compatibility (already the case for Zygote, ForwardDiff)
-- Toeplitz Matrices compatibility
+## Related Work
 
 Directly inspired by the [MLKernels](https://github.com/trthatcher/MLKernels.jl) package.
 
+See the JuliaGaussianProcesses [github organisation](https://github.com/JuliaGaussianProcesses) and [website](https://juliagaussianprocesses.github.io/) for more related packages.
```
Comment (theogf) [inline README.md:55]:
```suggestion
See the JuliaGaussianProcesses [Github organisation](https://github.com/JuliaGaussianProcesses) and [website](https://juliagaussianprocesses.github.io/) for more related packages.
```

**Review by theogf — APPROVED** [review]

**Comment by willtebbutt** [comment]
Thanks for the review @st-- @theogf @Crown421 . Will merge as-is -- please open another PR if there's anything else that needs changing.

**Pull request merged**